packages/api/lib/cwSfExecutionEventUtils.js

```diff
@@ -23,57 +25,89 @@ const executionStatusToWorkflowStatus = (executionStatus) => {
   return statusMap[executionStatus];
 };
 
+/**
+ * Searches the Execution step History for the TaskStateEntered pertaining to
+ * the failed task Id.  HistoryEvent ids are numbered sequentially, starting at
+ * one.
+*
+ * @param {HistoryEvent[]} events
+ * @param {number} lastFailureId
+ * @returns {string} name of the current stepfunction task or 'UnknownFailedStepName'.
+ */
+const getFailedStepName = (events, lastFailureId) => {
+  try {
+    const previousEvents = events.slice(0, lastFailureId - 1);
```

Comment:
Update L 40 to 
```
    const startEvents = events.filter((e) => e.type === 'TaskStateEntered');
```
and the units pass.

Reply:
I think it's possible that a task state may be entered after the failed event and This just drops all of those off, so that I can pop() the correct value off of the remaining task states.

Reply:
> Update L 40 to
> 
> ```
>     const startEvents = events.filter((e) => e.type === 'TaskStateEntered');
> ```
> 
> and the units pass.

I didn't know they didn't pass. Oops. But also, I think I want to only be looking through task states entered that occur before the failed step event

Reply:
`I think it's possible that a task state may be entered after the failed event` 

So we're considering a case like 

Task -> on failure -> another task -> cleanup task/choice/etc  	->  <Workflow Failed> ? 

Good point.  👍🏻  

In that case, we should have a unit test that explicitly covers that scenario.

Reply:
>and the units pass.

No, they pass as is, they *also* pass if you remove the line.     We need to add tests to cover the "workflow step fails, catches the failure and executes a cleanup path, *then* fails given the assertion (and I think it's correct)  that's a valid workflow path.

Reply:
This was addressed by f598582ecabdfd2bb165609f7ab0c6c7dbeaac0c